### PR TITLE
[CM H2] Remove kdb set functionality that creates a null key

### LINKED ISSUE #3571: Remove kdb set functionality that creates a null key.
In src/tools/kdb/set.cpp

`kdb set user:/tests/empty` should fail, it should require at least 2 arguments.

Make sure to also update documentation doc/help/kdb-set.md

(Follow up from #2540)

### PR DESCRIPTION
**Pull request by robaerd**
Closes #3571.

## Basics

These points need to be fulfilled for every PR:

- [x] Short descriptions of your changes are in the release notes
      (added as entry in `doc/news/_preparation_next_release.md` which
      contains `_(my name)_`)
      **Please always add something to the release notes.**
- [x] Details of what you changed are in commit messages
      (first line should have `module: short statement` syntax)
- [x] References to issues, e.g. `close #X`, are in the commit messages.
- [ ] The buildservers are happy. If not, fix **in this order**:
  - [ ] add a line in `doc/news/_preparation_next_release.md`
  - [ ] reformat the code with `scripts/dev/reformat-all`
  - [ ] make all unit tests pass
  - [ ] fix all memleaks
- [ ] The PR is rebased with current master.

If you have any troubles fulfilling these criteria, please write
about the trouble as comment in the PR. We will help you.
But we cannot accept PRs that do not fulfill the basics.

## Checklist

Check relevant points but **please do not remove entries**.
For docu fixes, spell checking, and similar none of these points below
need to be checked.

- [ ] I added unit tests for my code
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [ ] I added code comments, logging, and assertions as appropriate (see [Coding Guidelines](https://master.libelektra.org/doc/CODING.md))
- [ ] I updated all meta data (e.g. README.md of plugins and [METADATA.ini](https://master.libelektra.org/doc/METADATA.ini))
- [ ] I mentioned every code not directly written by me in [THIRD-PARTY-LICENSES](doc/THIRD-PARTY-LICENSES)

## Review

Reviewers will usually check the following:

- [ ] Documentation is introductory, concise, good to read and describes everything what the PR does
- [ ] Examples are well chosen and understandable
- [ ] Code is conforming to [our Coding Guidelines](https://master.libelektra.org/doc/CODING.md)
- [ ] APIs are conforming to [our Design Guidelines](https://master.libelektra.org/doc/DESIGN.md)
- [ ] Code is consistent to [our Design Decisions](https://master.libelektra.org/doc/decisions)

## Labels

If you are already Elektra developer:

- Add the "work in progress" label if you do not want the PR to be reviewed yet.
- Add the "ready to merge" label **if the basics are fulfilled** and you also
  say that everything is ready to be merged.

### CODE REVIEW

**Comment by robaerd** [comment]
@markus2330 @kodebach  I changed all `kdb set` operations where null keys were created, to create empty keys. But now the yamlcpp plugin markdown tests fail because the "___dirdata" value was added to each key that is not a leaf node, as it now contains an empty value.
Should I just simply update the tests or is a change in the yamlcpp plugin necessary?

**Comment by kodebach** [comment]
The test that files is `testshell_markdown_yamlcpp`. This test runs all the example shell scripts in [`yamlcpp`'s README](https://github.com/ElektraInitiative/libelektra/tree/master/src/plugins/yamlcpp). Some of the scripts use the two argument version of `kdb set`. You'll have to update the example scripts.

I don't think there is a need to change `yamlcpp`, but let's see what happens once you've changed the examples.

**Comment by markus2330** [comment]
@sanssecours Can you have a look at the question https://github.com/ElektraInitiative/libelektra/pull/3773#issuecomment-818618191

**Comment by sanssecours** [comment]
> @sanssecours Can you have a look at the question [#3773 (comment)](https://github.com/ElektraInitiative/libelektra/pull/3773#issuecomment-818618191) please?

I think the updates from Robert look good to me 👍. He also described the behavior of the plugin (in conjunction with the “Directory Value” plugin) quite well. Currently it treats `NULL` (array parent) keys as “empty” (or in other words as “nothing”). If there is a value stored in the array parent (even if its an empty string), then Directory Value will create a `___dirdata` node.

File: src/tools/kdb/set.hpp [hunk 1]
```diff
@@ -38,8 +38,7 @@ class SetCommand : public Command
 
 	virtual std::string getLongHelpText () override
 	{
-		return "If no value is given, it will be set to a null-value\n"
-		       "To get an empty value you need to quote like \"\" (depending on shell)\n"
+		return "To get an empty value you need to quote like \"\" (depending on shell)\n"
```
Comment (mpranj) [inline src/tools/kdb/set.hpp:41]: :see_no_evil: 
```suggestion
		return "To set an empty value you need to quote like \"\" (depending on shell)\n"
```

**Comment by mpranj** [comment]
Thank you so much!

**Pull request merged**